test/jdk/java/foreign/TestLibraryLookup.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @modules jdk.incubator.foreign/jdk.internal.foreign
+ * @run testng/othervm -Dforeign.restricted=permit TestLibraryLookup
+ */
+
+import jdk.incubator.foreign.LibraryLookup;
+import jdk.incubator.foreign.MemoryAddress;
+import jdk.internal.foreign.LibrariesHelper;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.testng.Assert.*;
+
+public class TestLibraryLookup {
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testInvalidLookupName() {
+        LibraryLookup.ofLibrary("NonExistent");
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testInvalidLookupPath() {
+        LibraryLookup.ofPath("/foo/bar/NonExistent");
+    }
```

Comment:
This test is failing on Windows because the path is not absolute. Suggestion:
```suggestion
    @Test(expectedExceptions = IllegalArgumentException.class)
    public void testInvalidLookupPath() {
        LibraryLookup.ofPath(Path.of("NonExistent").toAbsolutePath().toString());
    }
```